src/vehicle.cpp

```diff
@@ -1248,6 +1252,152 @@ int vehicle::install_part( int dx, int dy, const vehicle_part &new_part )
     return parts.size() - 1;
 }
 
+bool vehicle::find_rackable_vehicle( std::vector<std::vector<int>> racks_parts )
+{
+    // the racks are already sorted
+    for( auto rack_parts : racks_parts ) {
+        int found_vehicle = -1;
+        std::vector<vehicle *> carry_vehs;
+        carry_vehs.assign( 4, nullptr );
+        vehicle *test_veh = nullptr;
+        std::vector<std::set<tripoint>> test_pts;
+        std::set<tripoint> test_pt;
+        test_pts.assign( 4, test_pt );
+        for( auto rack_part : rack_parts ) {
+            tripoint rack_pos = global_part_pos3( rack_part );
+            for( int i = 0; i < 4; i++ ) {
+                tripoint search_pos( rack_pos + vehicles::cardinal_d[ i ] );
+                test_veh = veh_pointer_or_null( g->m.veh_at( search_pos ) );
+                if( test_veh == this ) {
+                    continue;
+                    // we had a candidate vehicle, this veh is different
+                } else if( carry_vehs[ i ] && test_veh != carry_vehs[ i ] ) {
+                    // is every part of the vehicle adjacent to the rack?
+                    if( test_pts[ i ] == carry_vehs[ i ]->occupied_points ) {
+                        found_vehicle = i;
+                        break;
+                    }
+                    // discard the old candidate vehicle
+                    carry_vehs[ i ] = nullptr;
+                    test_pts[ i ].clear();
+                }
+                if( test_veh && !carry_vehs[ i ] ) {
```

Comment:
Maybe move the check for test_veh == nullptr to the top so we bail out ASAP if there's no vehicle at that position?

Reply:
No, that would be wrong. The logic is if we have a potential carry vehicle, and the test vehicle isn't our vehicle or the potential carry vehicle, then we've reached the end of the potential carry vehicle, and it doesn't matter if the test vehicle is nullptr (so we're looking at empty space) or at a new vehicle (!nullptr).

I should reverse the logic to
```
if( test_veh == nullptr ) {
    continue;
} else if ( carry_vehs[ i ] == nullptr ) {
    carry_vehs[ i ] = test_veh;
}
```
for improved clarity.

Reply:
Fixed.